src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -907,28 +949,20 @@ def _reorder_cache(
 
         Output shares the same memory storage as `past`.
         """
-        batch_size_times_num_heads, head_dim, seq_length = past[0][0].shape
-        batch_size = len(beam_idx)
-        num_heads = batch_size_times_num_heads // batch_size
+        standardized_past = self._convert_to_standard_cache(past, batch_size=len(beam_idx))
+
         # Get a copy of `beam_idx` on all the devices where we need those indices.
         device_to_beam_idx = {
             past_state.device: beam_idx.to(past_state.device) for layer_past in past for past_state in layer_past
         }
-        # key: layer_past[0] [batch_size * num_heads, head_dim, seq_length]
-        # value: layer_past[1] [batch_size * num_heads, seq_length, head_dim]
-        return tuple(
+        reordered_past = tuple(
             (
-                layer_past[0]
-                .view(batch_size, num_heads, head_dim, seq_length)
-                .index_select(0, device_to_beam_idx[layer_past[0].device])
-                .view(batch_size_times_num_heads, head_dim, seq_length),
-                layer_past[1]
-                .view(batch_size, num_heads, seq_length, head_dim)
-                .index_select(0, device_to_beam_idx[layer_past[0].device])
-                .view(batch_size_times_num_heads, seq_length, head_dim),
+                layer_past[0].index_select(0, device_to_beam_idx[layer_past[0].device]),
+                layer_past[1].index_select(0, device_to_beam_idx[layer_past[0].device]),
```

Comment:
```suggestion
                layer_past[1].index_select(0, device_to_beam_idx[layer_past[1].device]),
```
Maybe this will fail in a multi-GPU setup? (thinking of BLOOM-176) but I am not sure didn't tested it

Reply:
I jus noticed that we were also doing `layer_past[1].view(batch_size, num_heads, seq_length, head_dim).index_select(0, device_to_beam_idx[layer_past[0].device])` before so probably what I propose is not needed!

Reply:
I know absolutely nothing about torch + multi GPU -- how could I test it?

Reply:
Here is a small snippet to try this out (ofc you need a multi-gpu setup):
```python
from transformers import AutoModelForCausalLM, AutoTokenizer


max_memory = {0:"2GB", 1:"3GB"}
model = AutoModelForCausalLM.from_pretrained("bigscience/bloom-1b7", device_map="auto", max_memory=max_memory, torch_dtype="auto")
# Confirm the model is on multiple GPUs
print(model.hf_device_map)
>>> {'transformer.word_embeddings': 0, 'lm_head': 0, 'transformer.word_embeddings_layernorm': 0, 'transformer.h.0': 0, 'transformer.h.1': 0, 'transformer.h.2': 0, 'transformer.h.3': 0, 'transformer.h.4': 0, 'transformer.h.5': 0, 'transformer.h.6': 0, 'transformer.h.7': 0, 'transformer.h.8': 1, 'transformer.h.9': 1, 'transformer.h.10': 1, 'transformer.h.11': 1, 'transformer.h.12': 1, 'transformer.h.13': 1, 'transformer.h.14': 1, 'transformer.h.15': 1, 'transformer.h.16': 1, 'transformer.h.17': 1, 'transformer.h.18': 1, 'transformer.h.19': 1, 'transformer.h.20': 1, 'transformer.h.21': 1, 'transformer.h.22': 1, 'transformer.h.23': 1, 'transformer.ln_f': 1}

input_text = "Twitter is "
input_ids = tokenizer(input_text, return_tensors="pt")["input_ids"].to(0)
out = model.generate(input_ids, penalty_alpha=0.6, top_k=4, max_length=512)
print(tokenizer.decode(out[0]))
>>> 'Twitter is  a social networking site that allows users to create and manage their own profiles, and to connect with friends and other users. The site is owned by Facebook Inc. and is operated by Facebook.</s>'
```
Seems to work like charm! I think that my comment can be safely ignored

Reply:
Btw `out = model.generate(input_ids, penalty_alpha=0.6, top_k=4, max_length=512)` is the correct way to try contrastive learning right?

Reply:
Yup, it is! Thanks for double-checking 🙌